docs: nomad secret role backend is required (#2024)
hashicorp/terraform-provider-vault: website/docs/r/nomad_secret_role.html.md

  The `namespace` is always relative to the provider's configured [namespace](/docs/providers/vault#namespace).
   *Available only for Vault Enterprise*.

> * `backend` - (Required) The unique path this backend should be mounted at. Must

Reviewer: nit: slight change in language since this is more consistent with other secret engines and backend is already mounted

```suggestion
* `backend` - (Required) The path where the Nomad secrets engine is mounted.
```

Also for what it's worth we do validate and trim trailing/leading slashes within the code, but I think it doesn't hurt to mention that there is no need for `/`

Author: Ah yes, you are correct. I am going to remove this then because it is technically incorrect
